tests/lib/fakeinstaller/main.go

```diff
@@ -0,0 +1,232 @@
+	if len(volumes) != 1 {
+		return fmt.Errorf("got unexpected number of volumes %v", len(volumes))
+	}
+
+	diskLayout, err := gadget.OnDiskVolumeFromDevice(bootDevice)
+	if err != nil {
+		return fmt.Errorf("cannot read %v partitions: %v", bootDevice, err)
+	}
+	if len(diskLayout.Structure) > 0 {
+		return fmt.Errorf("cannot yet install on a disk that has partitions")
+	}
+
+	constraints := gadget.LayoutConstraints{
+		// XXX: cargo-culted
+		NonMBRStartOffset: 1 * quantity.OffsetMiB,
+		// at this point we only care about creating partitions
+		SkipResolveContent:         true,
+		SkipLayoutStructureContent: true,
+	}
+
+	// FIXME: refactor gadget/install code to not take these dirs
+	gadgetRoot := ""
+	kernelRoot := ""
+
+	// TODO: support multiple volumes, see gadget/install/install.go
+	vol := firstVol(volumes)
+	lvol, err := gadget.LayoutVolume(gadgetRoot, kernelRoot, vol, constraints)
+	if err != nil {
+		return fmt.Errorf("cannot layout volume: %v", err)
+	}
+
+	iconst := &install.Constraints{AllPartitions: true}
+	created, err := install.CreateMissingPartitions(gadgetRoot, diskLayout, lvol, iconst)
+	if err != nil {
+		return fmt.Errorf("cannot create parititons: %v", err)
+	}
+	logger.Noticef("created %v partitions", created)
+
+	return nil
+}
+
+func runMntFor(label string) string {
+	return filepath.Join(dirs.GlobalRootDir, "/run/mnt/", label)
+}
+
+func postSystemsInstallSetupStorageEncryption(details *client.SystemDetails) error {
+	// TODO: check details.StorageEncryption and call POST
+	// /systems/<seed-label> with "action":"install" and
+	// "step":"setup-storage-encryption"
+	return nil
+}
+
+func postSystemsInstallFinish(details *client.SystemDetails) error {
+	// TODO: call POST /systems/<seed-label> with
+	// "action":"install" and "step":"finish"
+	return nil
+}
+
+// XXX: pass in created partitions instead?
+// XXX2: should we mount or will snapd mount?
+func createAndMountFilesystems(bootDevice string, volumes map[string]*gadget.Volume) error {
+	// only support a single volume for now
+	if len(volumes) != 1 {
+		return fmt.Errorf("got unexpected number of volumes %v", len(volumes))
+	}
+
+	disk, err := disks.DiskFromDeviceName(bootDevice)
+	if err != nil {
+		return err
+	}
+	vol := firstVol(volumes)
+
+	for _, stru := range vol.Structure {
+		if stru.Label == "" || stru.Filesystem == "" {
+			continue
+		}
+
+		part, err := disk.FindMatchingPartitionWithPartLabel(stru.Label)
+		if err != nil {
+			return err
+		}
+		// XXX: reuse
+		// gadget/install/content.go:mountFilesystem() instead
+		// (it will also call udevadm)
+		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+			return err
+		}
+
+		// mount
+		mountPoint := runMntFor(stru.Label)
+		if err := os.MkdirAll(mountPoint, 0755); err != nil {
+			return err
+		}
+		// XXX: is there a better way?
+		if output, err := exec.Command("mount", part.KernelDeviceNode, mountPoint).CombinedOutput(); err != nil {
```

Comment:
Maybe calling `systemd-mount`, but probably overkill at least while this is just a fake.